File: 0001. Two Sum/python/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diff = {}

        for i, value in enumerate(nums):
            found = target - value

            if found in diff:
                return [diff[found], i]

            diff[value] = i

        return [-1, -1]

File: 0001. Two Sum/c#/Solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var diff = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int value = nums[i];
            int found = target - value;

            if (diff.ContainsKey(found)) {
                return [diff[found], i];
            }
            diff[value] = i;
        }
        return [-1, -1];
    }
}
